packages/salesforcedx-vscode-apex-replay-debugger/src/breakpoints/checkpointService.ts

```diff
@@ -426,6 +422,166 @@ export class CheckpointService implements TreeDataProvider<BaseNode> {
       return false;
     }
   }
+
+  // The order of operations here should be to
+  // 1. Get the source/line information
+  // 2. Validate the existing checkpoint information
+  //    a. validate there are only 5 active checkpoints
+  //    b. validate that the active checkpoint information is correct
+  //    c. set the typeRef on each checkpoint (requires the source/line information)
+  // 3. Remove any existing checkpoints
+  // 4. Create the new checkpoints
+  public static async sfdxCreateCheckpoints(): Promise<boolean> {
+    // In-spite of waiting for the lock, we still want subsequent calls to immediately return
+    // from this if checkpoints are already being created instead of stacking them up.
+    if (!creatingCheckpoints) {
+      creatingCheckpoints = true;
+    } else {
+      return false;
+    }
+    let updateError = false;
+    // The status message isn't changing, call to localize it once and use the localized string in the
```

Comment:
A few comments about `sfdxCreateCheckpoints()`
* Most of the comments don't seem to be needed, as they comment what is obvious on the next line, and should be removed
* This function is pretty large, and it would be better if it was broken up into smaller functions
* Unit tests?

Reply:
Agreed on all points.  (and those made below).  I'll create a ticket and reference this PR.